src/gc/collector.cpp

```diff
@@ -115,8 +140,12 @@ class TraceStack {
             pop_chunk();
             assert(cur == end);
             return *--cur; // no need for any bounds checks here since we're guaranteed we're CHUNK_SIZE from the start
+        } else {
+            // We emptied the stack, but we should prepare a new chunk in case another item
```

Comment:
does this actually help?  it guarantees we have one additional `get_chunk()` call at the end of the last use of the trace stack.